templates/limitOrder.go

```diff
@@ -0,0 +1,125 @@
+package templates
+
+import (
+	"encoding/base64"
+	"github.com/algorand/go-algorand-sdk/crypto"
+	"github.com/algorand/go-algorand-sdk/transaction"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+// Split template representation
+type LimitOrder struct {
+	ContractTemplate
+	assetID uint64
+	owner   string
+}
+
+// GetSwapAssetsTransaction returns a group transaction array which transfer funds according to the contract's ratio
+// assetAmount: amount of assets to be sent
+// contract: byteform of the contract from the payer
+// secretKey: secret key for signing transactions
+// fee: fee per byte used for the transactions
+// algoAmount: number of algos to transfer
+// firstRound: first round on which these txns will be valid
+// lastRound: last round on which these txns will be valid
+// genesisHash: genesisHash indicating the network for the txns
+// the first payment sends money (Algos) from contract to the recipient (we'll call him Buyer), closing the rest of the account to Owner
+// the second payment sends money (the asset) from Buyer to the Owner
+// these transactions will be rejected if they do not meet the restrictions set by the contract
+func (lo LimitOrder) GetSwapAssetsTransaction(assetAmount uint64, contract, secretKey []byte, fee, algoAmount, firstRound, lastRound uint64, genesisHash []byte) ([]byte, error) {
```

Comment:
This one does the correct thing!

Reply:
we should probably rename `algoAmount` to `microAlgoAmount`